src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/hex/Rebar3TreeParser.groovy

```diff
@@ -0,0 +1,159 @@
+                    Dependency currentDependency = createDependencyFromLine(line)
+
+                    final int currentTreeLevel = getDependencyLevelFromLine(line)
+
+                    if (currentTreeLevel == previousTreeLevel + 1) {
+                        dependencyStack.push(previousDependency)
+                    } else if (currentTreeLevel < previousTreeLevel) {
+                        int levelDelta = (previousTreeLevel - currentTreeLevel)
+                        for (int levels = 0; levels < levelDelta; levels++) {
+                            dependencyStack.pop()
+                        }
+                    } else if (currentTreeLevel != previousTreeLevel) {
+                        logger.error(String.format('The tree level (%s) and this line (%s) with count %s can\'t be reconciled.', previousTreeLevel, line, currentTreeLevel))
+                    }
+
+                    if (dependencyStack.size() == 0) {
+                        if ( isProject(line) ) {
+                            projectName = currentDependency.name
+                            projectVersionName = currentDependency.version
+                        } else {
+                            graph.addChildToRoot(currentDependency)
+                        }
+                    } else if (dependencyStack.size() == 1 && dependencyStack.peek().name.equals(projectName) && dependencyStack.peek().version.equals(projectVersionName)) {
+                        graph.addChildToRoot(currentDependency)
+                    } else {
+                        graph.addChildWithParents(currentDependency, dependencyStack.peek())
+                    }
+
+                    previousDependency = currentDependency
+                    previousTreeLevel = currentTreeLevel
+                }
+            }
+        } catch (final Exception e) {
+            logger.error('Exception parsing rebar output: ' + e.getMessage())
+        }
+        detectProject.setProjectNameIfNotSet(projectName)
+        detectProject.setProjectVersionNameIfNotSet(projectVersionName)
+
+        final ExternalId id = externalIdFactory.createNameVersionExternalId(Forge.HEX, projectName, projectVersionName)
+
+        return new DetectCodeLocation(BomToolType.HEX, sourcePath, projectName, projectVersionName, id, graph)
+    }
+
+    private Dependency createDependencyFromLine(String line) {
+        String nameVersionLine = reduceLineToNameVersion(line)
+        String name = nameVersionLine.substring(0, nameVersionLine.lastIndexOf(HORIZONTAL_SEPARATOR_CHARACTER))
+        String version = nameVersionLine.substring(nameVersionLine.lastIndexOf(HORIZONTAL_SEPARATOR_CHARACTER) + 1)
+        ExternalId externalId  = externalIdFactory.createNameVersionExternalId(Forge.HEX, name, version)
+
+        return new Dependency(name, version, externalId)
+    }
+
+    private String reduceLineToNameVersion(String line) {
+        for (String specialCharacter : [
```

Comment:
Prefer specialCharacters = [] and then for (String specialCharacter : specialCharacters) instead of this. Why not make Special characters a static?